src/shared/components/query/CaseSetSelectorUtils.ts

```diff
@@ -0,0 +1,57 @@
+import * as _ from "lodash";
+
+export enum CaseSetId {
+	"custom"       = '-1',
+	"mutation_cna" = 'w_mut_cna',
+	"mutation"     = 'w_mut',
+	'cna'          = 'w_cna',
+	'all'          = 'all'
+}
+
+export type CustomCaseSet = {
+	name: string,
+	description:string,
+	value: CaseSetId,
+	isdefault: boolean
+};
+
+export const CustomCaseSets: CustomCaseSet[] = [ 
+	{name: 'All', description: 'All cases in the selected cohorts', value: CaseSetId.all, isdefault : false} ,
+	{name: 'Cases with mutations data', description: 'All cases with mutations data', value: CaseSetId.mutation, isdefault : false},
+	{name: 'Cases with copy number alterations data', description: 'All cases with copy number alterations data', value: CaseSetId.cna, isdefault : false},
+	{name: 'Cases with both mutations and copy number alterations data', description: 'All cases with both mutations and copy number alterations data', value: CaseSetId.mutation_cna, isdefault : false},
+	{name: 'User-defined Case List', description: 'Specify your own case list', value: CaseSetId.custom, isdefault : true}
+]
+
+export function getFilteredCustomCaseSets(isVirtualStudy:boolean, profiledSamplesCount: { mutation: number; cna: number; mutation_cna: number; all: number}) {
+    return _.reduce(CustomCaseSets, (acc: CustomCaseSet[], next) => {
+        if (next.isdefault) {
+            acc.push(next)
+        } else if (isVirtualStudy) {
+            let count = 0;
+            switch(next.value) {
```

Comment:
you don't have to change this but wouldn't this be more simply written:
profiledSamplesCount.mutation.length || profiledSamplesCount.cna.length || ... 
No switch statement, no count variable.

Reply:
Actually I need the `count` here. It is append to the sample list name at line 52.

Reply:
`value` is of type enum
```
enum CaseSetId {
	"custom"       = '-1',
	"mutation_cna" = 'w_mut_cna',
	"mutation"     = 'w_mut',
	'cna'          = 'w_cna',
	'all'          = 'all'
}
```
Then I need to change `profiledSamplesCount` to `{w_mutation:number, w_cna:number, w_mutation_cna:number, all:number}`